### PR TITLE
Add media description feature using Azure Content Understanding

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This PR adds a new optional feature that will extract figures in documents (using Azure Document Intelligence) figures output mode and send those figures to Azure Content Understanding (a new service that uses multimodal models) to generate a figure description. It will then insert that figure description into the content, which will then get sent for chunking. If the figure is of a graph or chart, it will include an HTML table with the data.
This gives developers a more lightweight approach to ingest media-rich documents and can be compared to the more heavyweight GPT-4-vision approach.

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [x] The current tests all pass (`python -m pytest`).
- [x] I added tests that prove my fix is effective or that my feature works
- [x] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [x] I ran `python -m mypy` to check for type errors
- [X] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

**Comment by pamelafox** [comment]
I need to do some cleanup, per the CI. This also needs a few more tests, as it currently only has a test for the changes to the splitting algorithm.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -166,6 +166,8 @@ If you followed the steps above to install the pre-commit hooks, then you can ju
 
 When adding new azd environment variables, please remember to update:
 
+1. [main.parameters.json](./main.parameters.json)
+1. [appEnvVariables in main.bicep](./main.bicep)
```
Comment (mattgotteiner) [inline CONTRIBUTING.md:170]: infra?

File: app/backend/prepdocslib/figure_output.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
+"figures": [
```
Comment (mattgotteiner) [inline app/backend/prepdocslib/figure_output.json:1]: should we pull this into the data directory? or not check this in at all?

File: app/backend/prepdocslib/pdfparser.py [hunk 3]
```diff
@@ -107,5 +198,28 @@ def table_to_html(cls, table: DocumentTable):
...
+        :param page_number: The page number to crop from (0-indexed).
+        :param bounding_box: A tuple of (x0, y0, x1, y1) coordinates for the bounding box.
+        :return: A PIL Image of the cropped area.
+        """
+        page = doc.load_page(page_number)
+
+        # Cropping the page. The rect requires the coordinates in the format (x0, y0, x1, y1).
+        bbx = [x * 72 for x in bounding_box]
+        rect = pymupdf.Rect(bbx)
+        # 72 is the DPI ? what? explain this from CU
```
Comment (mattgotteiner) [inline app/backend/prepdocslib/pdfparser.py:219]: let's fill this in:

Bound box is scaled to 72 dots per inch
Assume the PDF has 300 DPI
The matrix is used to convert between these 2 units

**Review by mattgotteiner — APPROVED** [review]

**Comment by pamelafox** [comment]
I have now completed both unit testing and manual testing, and this is good to merge. We will merge/release on Monday.

**Pull request merged**